node/src/main/scala/org/bitcoins/node/Node.scala

```diff
@@ -45,30 +44,117 @@ trait Node extends NodeApi with ChainQueryApi with P2PLogger {
 
   implicit def executionContext: ExecutionContext = system.dispatcher
 
-  val peers: Vector[Peer]
-
-  private val _peerServices: mutable.Map[Peer, ServiceIdentifier] =
+  private val _peerData: mutable.Map[Peer, PeerData] =
```

Comment:
General organization thoughts: 

Let's move all this peer stuff out of the `Node` trait. It probably should all go into something called `PeerManager`. Then this trait can just have one reference to `PeerManager` and call methods on it.